### PR TITLE
[WIP] Adds a currently adminspawn only book for making people headrevs.

### PR DESCRIPTION
**Pull request by F-OS**
Almost done, should only need a bit of code to finish it up.
why: Just think having a more RP-Friendly way to make a headrev like the ling vial would be good.
:cl:
 * content: Adds a adminspawn only revolutionary conversion book

### CODE REVIEW

**Comment by F-OS** [comment]
eeee byond

**Comment by F-OS** [comment]
ok i still need to fix this so uhh sorry agh

File: code/game/objects/items/rev_book.dm [hunk 1]
```diff
@@ -0,0 +1,49 @@
...
+        var/mob/living/carbon/human/H = user
+        if(H.mind)
+            var/datum/mind/M = H.mind
+            if(!isrevolutionary(H))
+                user.visible_message("<span class='notice'>[user] opens [src.name] and starts reading intently.</span>","<span class='notice'>You open [src.name] and start reading intently.</span>")
+                sleep(50)
+                to_chat(H, "<span class='notice'>[pick("You suddenly realize how oppressed you really are under NT", "You start to question your allegiance to NT", "You wonder to yourself, what exactly happened to KC13?", "You suddenly realize how little freedom you have here", "You read something that deeply resonates with you and makes you question your beliefs")].</span>")
+                sleep(100)
+                to_chat(H, "<span class='notice'>[pick("Wait, they can't really be behind the lunar bombings of 2100, can they?", "This isn't the NT I joined.", "Oh god, they really did that, all those people, gone, and for what, some plasma?", "Dear lord, who have I been working for?", "The syndicate may be bad, but this is worse."]</span>"))
```
Comment (ShiftyRail) [inline code/game/objects/items/rev_book.dm]: You forgot to close the parenthesis in of your `pick`, it needs to be closed inside the brackets

File: code/game/objects/items/rev_book.dm [hunk 2]
```diff
@@ -0,0 +1,49 @@
...
+                sleep(10)
+                to_chat(H, "<span class='notice'>You renounce your allegiance to nanotransen and decide to join the fight against corporate tyranny, oppression, and persecution. Death to the capitalist oppressors, Vive la révolution!</span>")
+                var/wikiroute = role_wiki[ROLE_REVOLUTIONARY]
+                to_chat(H, "<span class='info'><a HREF='?src=\ref[H];getwiki=[wikiroute]'>(Wiki Guide)</a></span>")
+                //wip requires coder help
+                if(prob(25))
+                    H.nutrition = 10 // le holodomor memes
+                uses = uses - 1
+                if(uses = 0 && infinite_uses != 1)
+                    user.visible_message("<span class='notice'>[src.name] is engulfed in a blue light and vanishes!</span>")
```
Comment (ShiftyRail) [inline code/game/objects/items/rev_book.dm]: `\The [src]`

File: code/game/objects/items/rev_book.dm [hunk 3]
```diff
@@ -0,0 +1,49 @@
...
+                to_chat(H, "<span class='notice'>[pick("Wait, they can't really be behind the lunar bombings of 2100, can they?", "This isn't the NT I joined.", "Oh god, they really did that, all those people, gone, and for what, some plasma?", "Dear lord, who have I been working for?", "The syndicate may be bad, but this is worse."]</span>"))
+                sleep(10)
+                to_chat(H, "<span class='notice'>You renounce your allegiance to nanotransen and decide to join the fight against corporate tyranny, oppression, and persecution. Death to the capitalist oppressors, Vive la révolution!</span>")
+                var/wikiroute = role_wiki[ROLE_REVOLUTIONARY]
+                to_chat(H, "<span class='info'><a HREF='?src=\ref[H];getwiki=[wikiroute]'>(Wiki Guide)</a></span>")
+                //wip requires coder help
+                if(prob(25))
+                    H.nutrition = 10 // le holodomor memes
+                uses = uses - 1
+                if(uses = 0 && infinite_uses != 1)
```
Comment (ShiftyRail) [inline code/game/objects/items/rev_book.dm]: `if(uses == 0)`, else it's an assignment and not a comparison

File: code/game/objects/items/rev_book.dm [hunk 4]
```diff
@@ -0,0 +1,49 @@
...
+    if(prob(10))
+        name = "A Guide to Becoming Poor and Hungry: 3rd edition"
+    else
+        name = pick("The Modern Corporate State and Revolution", "What I Saw: Memoirs of a Corporate Slave", "The Communist Manifesto: With Notes from Cyndy Kate", "The Theory and Practice of Corporate Totalitarianism", "The Conquest of Bread: A Syndicate Reprint")
+    ..()
+/obj/item/rev_book/attack_self(mob/user as mob)
+    for(var/obj/item/weapon/implant/loyalty/L in mob) // check loyalty implant in the contents
+        if(L.imp_in == H) // a check if it's actually implanted
+            to_chat("<span class='danger'>As you start to open the book, you feel a crippling pain in your head!</span>")
+            mob.AdjustStunned(10)
```
Comment (ShiftyRail) [inline code/game/objects/items/rev_book.dm]: It's `user` you want, and `user.Stun`

File: code/game/objects/items/rev_book.dm [hunk 5]
```diff
@@ -0,0 +1,49 @@
...
+    infinite_uses = 0
+    uses = 1
+/obj/item/rev_book/New()
+    if(prob(10))
+        name = "A Guide to Becoming Poor and Hungry: 3rd edition"
+    else
+        name = pick("The Modern Corporate State and Revolution", "What I Saw: Memoirs of a Corporate Slave", "The Communist Manifesto: With Notes from Cyndy Kate", "The Theory and Practice of Corporate Totalitarianism", "The Conquest of Bread: A Syndicate Reprint")
+    ..()
+/obj/item/rev_book/attack_self(mob/user as mob)
+    for(var/obj/item/weapon/implant/loyalty/L in mob) // check loyalty implant in the contents
```
Comment (ShiftyRail) [inline code/game/objects/items/rev_book.dm]: It's `user` you want

File: code/game/objects/items/rev_book.dm [hunk 6]
```diff
@@ -0,0 +1,49 @@
...
+    w_class = 1
+    infinite_uses = 0
+    uses = 1
+/obj/item/rev_book/New()
+    if(prob(10))
+        name = "A Guide to Becoming Poor and Hungry: 3rd edition"
+    else
+        name = pick("The Modern Corporate State and Revolution", "What I Saw: Memoirs of a Corporate Slave", "The Communist Manifesto: With Notes from Cyndy Kate", "The Theory and Practice of Corporate Totalitarianism", "The Conquest of Bread: A Syndicate Reprint")
+    ..()
+/obj/item/rev_book/attack_self(mob/user as mob)
```
Comment (ShiftyRail) [inline code/game/objects/items/rev_book.dm]: `as mob` is not needed

**Comment by F-OS** [comment]
@ShiftyRail By god, thank you so much, i'm not that competent at coding and thanks to your help i should be able to fix this PR.

**Comment by F-OS** [comment]
please just kill me now

File: code/game/objects/items/rev_book.dm [hunk 7]
```diff
@@ -0,0 +1,53 @@
...
+        var/mob/living/carbon/human/H = user
+        if(H.mind)
+            var/datum/mind/M = H.mind
+            if(!isrevolutionary(H))
+                user.visible_message("<span class='notice'>[user] opens [src.name] and starts reading intently.</span>","<span class='notice'>You open [src.name] and start reading intently.</span>")
+                sleep(50)
+                to_chat(H, "<span class='notice'>[pick("You suddenly realize how oppressed you really are under NT", "You start to question your allegiance to NT", "You wonder to yourself, what exactly happened to KC13?", "You suddenly realize how little freedom you have here", "You read something that deeply resonates with you and makes you question your beliefs")].</span>")
+                sleep(100)
+                to_chat(H, "<span class='notice'>[pick("Wait, they can't really be behind the lunar bombings of 2100, can they?", "This isn't the NT I joined.", "Oh god, they really did that, all those people, gone, and for what, some plasma?", "Dear lord, who have I been working for?", "The syndicate may be bad, but this is worse.")]</span>"))
```
Comment (ShiftyRail) [inline code/game/objects/items/rev_book.dm]: Gee Bill, two parenthesis ?

File: code/game/objects/items/rev_book.dm [hunk 8]
```diff
@@ -0,0 +1,53 @@
...
+                sleep(10)
+                to_chat(H, "<span class='notice'>You renounce your allegiance to nanotransen and decide to join the fight against corporate tyranny, oppression, and persecution. Death to the capitalist oppressors, Vive la révolution!</span>")
+                var/wikiroute = role_wiki[ROLE_REVOLUTIONARY]
+                to_chat(H, "<span class='info'><a HREF='?src=\ref[H];getwiki=[wikiroute]'>(Wiki Guide)</a></span>")
+                ticker.mode.head_revolutionaries += M.mind
```
Comment (ShiftyRail) [inline code/game/objects/items/rev_book.dm]: Check if your man still still has a mind.

File: code/game/objects/items/rev_book.dm [hunk 9]
```diff
@@ -0,0 +1,53 @@
...
+    else
+        name = pick("The Modern Corporate State and Revolution", "What I Saw: Memoirs of a Corporate Slave", "The Communist Manifesto: With Notes from Cyndy Kate", "The Theory and Practice of Corporate Totalitarianism", "The Conquest of Bread: A Syndicate Reprint")
+    ..()
+/obj/item/rev_book/attack_self(mob/user)
+    for(var/obj/item/weapon/implant/loyalty/L in user) // check loyalty implant in the contents
+        if(L.imp_in == user) // a check if it's actually implanted
+            to_chat("<span class='danger'>As you start to open the book, you feel a crippling pain in your head!</span>")
+            user.Stun = 10
+            return 0
+    if(istype(user, /mob/living/carbon/human) && !(jobban_isbanned(user, "Syndicate") || jobban_isbanned(user, "revolutionary")))
```
Comment (ShiftyRail) [inline code/game/objects/items/rev_book.dm]: `istype(user, /mob/living/carbon/human/)` => `ishuman(user)`

File: code/game/objects/items/rev_book.dm [hunk 10]
```diff
@@ -0,0 +1,53 @@
...
+    if(prob(10))
+        name = "A Guide to Becoming Poor and Hungry: 3rd edition"
+    else
+        name = pick("The Modern Corporate State and Revolution", "What I Saw: Memoirs of a Corporate Slave", "The Communist Manifesto: With Notes from Cyndy Kate", "The Theory and Practice of Corporate Totalitarianism", "The Conquest of Bread: A Syndicate Reprint")
+    ..()
+/obj/item/rev_book/attack_self(mob/user)
+    for(var/obj/item/weapon/implant/loyalty/L in user) // check loyalty implant in the contents
+        if(L.imp_in == user) // a check if it's actually implanted
+            to_chat("<span class='danger'>As you start to open the book, you feel a crippling pain in your head!</span>")
+            user.Stun = 10
```
Comment (ShiftyRail) [inline code/game/objects/items/rev_book.dm]: `user.Stun(10)` instead.

File: code/game/objects/items/rev_book.dm [hunk 11]
```diff
@@ -0,0 +1,53 @@
...
+                to_chat(H, "<span class='info'><a HREF='?src=\ref[H];getwiki=[wikiroute]'>(Wiki Guide)</a></span>")
+                ticker.mode.head_revolutionaries += M.mind
+                ticker.mode.update_rev_icons_added(M.mind)
+                M.mind.special_role = "Head Revolutionary"
+                ticker.mode.forge_revolutionary_objectives(M.mind)    
+                M.equip_to_slot_or_del(new/obj/item/device/flash, slot_l_store)
+                if(prob(25))
+                    H.nutrition = 10 // le holodomor memes
+                uses = uses - 1
+                if(uses == 0 && infinite_uses != 1)
```
Comment (ShiftyRail) [inline code/game/objects/items/rev_book.dm]: `if (!uses && !infinite_uses)` works just as well

File: code/game/objects/items/rev_book.dm [hunk 12]
```diff
@@ -0,0 +1,53 @@
...
+                    user.visible_message("<span class='notice'>[\The [src] is engulfed in a blue light and vanishes!</span>")
```
Comment (ShiftyRail) [inline code/game/objects/items/rev_book.dm]: What is this [ doing here ?

Comment (F-OS) [reply code/game/objects/items/rev_book.dm]: FUCK

**Comment by F-OS** [comment]
baby don't hurt me, don't hurt me no more

**Comment by Mothblocks** [comment]
captain reads the book and becomes a revolutionary (add head checks)

**Comment by F-OS** [comment]
@Jared-Fogle I will once i regain interest in this project, probably sometime this or next month.

**Comment by F-OS** [comment]
Stale. If this is really wanted pm me on discord @fosstarreagan#1440

**Pull request closed without merging**